### PR TITLE
feat: SEO metadata

### LINKED ISSUE #12: SEO Metadata

### PR DESCRIPTION
**Pull request by kylemocode**
## Changes made:
- close #12 
- add set metadata title and description to each pages
- the title will be different for each pages, ex:
  - 首頁 - The Mentorship 曼陀號領航計畫
  - 常見 FAQ - The Mentorship 曼陀號領航計畫
- add sitemap
- add robots.txt
- add favicon

截圖 2024-12-19 下午4 31 05

(the open graph URL will be added in the future)
## Test Scope / Change impact:

-

### CODE REVIEW

File: nextjs-app/app/faq/page.tsx [hunk 1]
```diff
@@ -1,3 +1,9 @@
+import type { Metadata } from "next";
+
+export const metadata: Metadata = {
+  title: "常見 FAQ",
```
Comment (gongtzuuuu) [inline nextjs-app/app/faq/page.tsx:4]: Perhaps the title could be either '常見問題' or 'FAQ.' Adding '常見' in front of 'FAQ' feels a bit repetitive. What do you think

File: nextjs-app/app/introduction/page.tsx [hunk 2]
```diff
@@ -1,3 +1,9 @@
+import type { Metadata } from "next";
+
+export const metadata: Metadata = {
+  title: "活動介紹",
```
Comment (gongtzuuuu) [inline nextjs-app/app/introduction/page.tsx:4]: Nitpick: Not sure if this is an introduction to the program or the series of events. If it's the former, '計畫介紹' might be more appropriate. What do you think?

File: nextjs-app/app/sitemap.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { MetadataRoute } from "next";
+
+export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
+  const baseUrl =
+    process.env.NEXT_PUBLIC_BASE_URL ?? "https://https://thementorship.tw/";
```
Comment (gongtzuuuu) [inline nextjs-app/app/sitemap.ts:5]: Seems that the default base URL repeats `https://` twice; it should be `https://thementorship.tw/`.

File: nextjs-app/app/robots.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { MetadataRoute } from "next";
+
+export default function robots(): MetadataRoute.Robots {
+  const baseUrl =
+    process.env.NEXT_PUBLIC_BASE_URL ?? "https://https://thementorship.tw/";
```
[the remark this answers is not in the document]
Comment (ChengYiLin) [reply nextjs-app/app/robots.ts:5]: I think the trailing `/` in the URL is unnecessary. 
It should be `https://thementorship.tw`.

**Review by ChengYiLin — APPROVED** [review]
LGTM~

File: nextjs-app/app/faq/page.tsx (hunk 1 above)
Comment (kylemocode) [reply nextjs-app/app/faq/page.tsx:4]: Confirmed with PM, we will use “常見問題”, thanks

File: nextjs-app/app/sitemap.ts (hunk 3 above)
Comment (kylemocode) [reply nextjs-app/app/sitemap.ts]: PM will stick to use original one

File: nextjs-app/app/introduction/page.tsx (hunk 2 above)
Comment (kylemocode) [reply nextjs-app/app/introduction/page.tsx:4]: PM will stick to use original one

**Pull request merged**